src/Prototype/Exe/Form/Signup.hs

```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE DeriveAnyClass #-}
+module Prototype.Exe.Form.Signup
+  ( Input
+  , Page(..)
+  , ResultPage(..)
+  ) where
+
+import           Smart.Html.Dsl                 ( HtmlCanvas )
+import qualified Smart.Html.Dsl                as Dsl
+import qualified Smart.Html.Render             as Render
+import           Smart.Html.Shared.Html.Icons
+import           Text.Blaze                     ( customAttribute )
+import qualified Text.Blaze.Html5              as H
+import           Text.Blaze.Html5               ( (!) )
+import qualified Text.Blaze.Html5.Attributes   as A
+import           Web.FormUrlEncoded             ( FromForm(..)
+                                                , parseUnique
+                                                )
+
+
+--------------------------------------------------------------------------------
+data Input = Input
+  { username     :: Text
+  , emailAddress :: Text
+  , password     :: Text
+  , iUnderstand  :: Text -- TODO Bool
+  }
```

Comment:
Since we like using Lenses (at least I do :-P), can we use lensy names for fields? It makes deriving lenses slightly less boilerplatey; eg. 

``` haskell
data Input = Input { _iUsername :: Text ... } 
```
etc. 

The small compromise, if it can even be considered a compromise will be that we'll need to use the same field names within HTML. I think that is fine. 
However; we can also use `makeLensesWith`; which will be more boilerplate.

Reply:
Also, we should be using `Password` for password fields.